frontend-react/cypress/support/commands.js

```diff
@@ -1,78 +1,67 @@
+    // log in with username and password
     cy.request("POST", "https://hhs-prime.oktapreview.com/api/v1/authn", {
         username,
         password,
         options: {
-            multiOptionalFactorEnroll: true,
+            multiOptionalFactorEnroll: false,
             warnBeforePasswordExpired: true,
         },
-    }).then((response) => {
-        const stateToken = response.body.stateToken;
-        const factorId = response.body._embedded.factors[0].id;
-        // const passCode = authenticator.generate(Cypress.env("okta_secret"));
-        const passCode = authenticator.generate("secret");
-        cy.request(
-            "POST",
-            `https://hhs-prime.oktapreview.com/api/v1/authn/factors/${factorId}/verify`,
-            {
-                passCode,
-                stateToken,
-            }
-        ).then((response) => {
+    })
+        // pass MFA
+        .then((response) => {
+            const stateToken = response.body.stateToken;
+            const factorId = response.body._embedded.factors[0].id;
+            const secret = Cypress.env("okta_secret");
+            const passCode = authenticator.generate(secret);
+            return cy.request(
+                "POST",
+                `https://hhs-prime.oktapreview.com/api/v1/authn/factors/${factorId}/verify`,
+                {
+                    passCode,
+                    stateToken,
+                }
+            );
+        })
+        // authorize
+        .then((response) => {
             const sessionToken = response.body.sessionToken;
-            cy.request(
-                "GET",
-                `https://hhs-prime.oktapreview.com/oauth2/default/v1/authorize?client_id=${Cypress.env('okta_client_id')}&redirect_uri=http%3A%2F%2Flocalhost.reportstream.gov%2F&response_type=token%20id_token&scope=openid%20email&nonce=thisisnotsafe&state=thisisbogus&sessionToken=${sessionToken}`
-            ).then((response) => {
-                const redirect = response.redirects[0];
-                const idTokenRegex = new RegExp(
-                    "(?:id_token=)((.[\\s\\S]*))(?:&access_token=)",
-                    "ig"
-                );
-                const accessTokenRegex = new RegExp(
-                    "(?:access_token=)((.[\\s\\S]*))(?:&token_type=Bearer)",
-                    "ig"
-                );
-                const id_token = idTokenRegex.exec(redirect)[1];
-                const access_token = accessTokenRegex.exec(redirect)[1];
-                cy.task("setAuth", {id_token, access_token});
-                cy.setLocalStorage("id_token", id_token);
-                cy.setLocalStorage("access_token", access_token);
-                cy.saveLocalStorage();
+            const authClient = new OktaAuth(oktaAuthConfig);
+            return authClient.token.getWithoutPrompt({
+                sessionToken,
             });
+        })
+        // set tokens in storage
+        .then(({ tokens }) => {
```

Comment:
Are these actions set up to mimic our behavior in-app in any way, and if so, should they somehow be safely/accurately kept congruent?

Reply:
the only real concern (at least for me) here is whether it would log the user in such a way that the user can navigate to auth walled parts of the site.

if we want to test the actual auth flow as a user would do that, I think we would want to write a test case specifically for that